csrc/type.cpp

```diff
@@ -111,12 +111,14 @@ bool isInclusiveType(const DataType& base_type, const DataType& wider_type) {
   if ((wider_type == DataType::Double ||
        wider_type == DataType::ComplexDouble) &&
       (base_type == DataType::Double || base_type == DataType::Float ||
-       base_type == DataType::Half || base_type == DataType::BFloat16)) {
+       base_type == DataType::Half || base_type == DataType::BFloat16 ||
+       base_type == DataType::Float8_e4m3fn || base_type == DataType::Float8_e5m2)) {
     return true;
   }
   if ((wider_type == DataType::Float || wider_type == DataType::ComplexFloat) &&
       (base_type == DataType::Float || base_type == DataType::Half ||
-       base_type == DataType::BFloat16)) {
+       base_type == DataType::BFloat16 ||
+       base_type == DataType::Float8_e4m3fn || base_type == DataType::Float8_e5m2)) {
```

Comment:
Do we also need a case where the wider type is 16-bit and the base type is 8-bit?